lib/velocy_pack/decoder.ex

```diff
@@ -267,15 +362,22 @@ defmodule VelocyPack.Decoder do
 
   @spec parse_object(integer(), binary()) :: {map(), binary()}
   defp parse_object(type, data) do
-    size_bytes = 1 <<< (type - 0x0b)
+    size_bytes = 1 <<< (type - 0x0B)
+
     <<total_size::integer-unsigned-little-unit(8)-size(size_bytes),
-      length::integer-unsigned-little-unit(8)-size(size_bytes),
-      rest::binary>> = data
+      length::integer-unsigned-little-unit(8)-size(size_bytes), rest::binary>> = data
+
     data_size = total_size - 1 - 2 * size_bytes
     <<data::binary-size(data_size), rest::binary>> = rest
-    index_table_size = if (length == 1), do: 0, else: length * size_bytes
-    {obj, <<_index_table::binary-size(index_table_size)>>} = parse_object_members(length, %{}, skip_zeros(data))
-    {obj, rest}
+    index_table_size = if length == 1, do: 0, else: length * size_bytes
+
+    case parse_object_members(length, %{}, skip_zeros(data)) do
+      {obj, <<_index_table::binary-size(index_table_size)>>} ->
+        {obj, rest}
+
+      {obj, <<3>>} ->
+        {obj, rest}
```

Comment:
What is this case for?

Reply:
This is one of those bandaids, I didn’t take the time to properly refactor everything. All objects with a single key/value pair have to end with a 0x3.

Reply:
Got it. I will dig through the VelocyPack documentation and code to verify if/why this is necessary and add an according unittest.

Reply:
This `3` would be the single entry of the index table. But since an index table makes no sense for a single object entry, this can be omitted. I am not sure if this is officially documented somewhere, but at least the Java implementation also has this optimization: https://github.com/arangodb/java-velocypack/blob/543f24419300fb12be6cc1b676089c924c0bd6f8/src/test/java/com/arangodb/velocypack/VPackSliceTest.java#L953-L960

Did you run into specific problems that you had to add this?

Reply:
Yea, arango 3.5 was responding with an invalid vpack error. 😐

Reply:
Interesting... I will look into this!

Reply:
Ok, so all (non-compact) objects have to have an index table: https://github.com/arangodb/velocypack/issues/45#issuecomment-539414346

The referenced unittest mislead me, so I specifically added special cases to omit the index table for objects with a single entry. :smile: Will fix that!

Reply:
Are the varying object types a matter of configuration or user preference?

Reply:
Compact objects have no index table, so as the name suggests, they are smaller (especially if the object has a lot of entries). However, if you want to get a particular entry, you have to perform a linear search. If an object has an index table we can perform a binary search. So whether compact or non-compact objects are better depends on your use case. At the moment every map is encoded as a non-compact object with an index table, but I have plans to make this configurable.

Reply:
Fixed